packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -239,13 +297,17 @@ export const plugin = <TContext>(
           // See comment above for why `didEnd` must be called in two hooks.
           didEnd(requestContext);
         },
-
-        didEncounterErrors(requestContext) {
+        async didEncounterErrors(requestContext) {
           // Search above for a comment about "didResolveSource" to see which
           // of the pre-source-resolution errors we are intentionally avoiding.
-          if (!didResolveSource) return;
+          if (!didResolveSource || endDone) return;
           treeBuilder.didEncounterErrors(requestContext.errors);
 
+          if (shouldReportTrace == null) {
```

Comment:
Following up on my comment above about the typing of `shouldReportTrace`:

I don't see when our own code would be setting `shouldReportTrace`` to `null` right now so I believe the intent of this check is for if it's undefined (which, yes, `null == undefined` but `null !== undefined`).

I think the intent is more clear if we just check for `undefined`?

```suggestion
          if (shouldReportTrace === undefined) {
```

Reply:
Correct, or we can also just make shouldTraceOperation fine to be called multiple times and bail out early if it's already been called.

Reply:
Removed this check and made `shouldTraceOperation` bail early if it has already been called.